AgentDiscoveries-Frontend/app/src/components/app.jsx

```diff
@@ -18,36 +18,42 @@ import LocationForm from './admin/location-form';
+    constructor(props) {
+        super(props);
+        this.state = {
+            isLoggedIn: isLoggedIn(),
+            isAdmin: isAdmin()
+        };
+        this.onLogInEvent = this.onLogInEvent.bind(this);
+        this.handleLogOut = this.handleLogOut.bind(this);
+    }
+
+    componentDidMount() {
+        window.addEventListener('login', this.onLogInEvent);
+    }
+
+    componentWillUnmount() {
+        window.removeEventListener('login', this.onLogInEvent);
+    }
+
+    onLogInEvent() {
+        this.setState({
+            isLoggedIn: isLoggedIn(),
+            isAdmin: isAdmin()
+        });
+    }
+
     render() {
         return (
             <React.Fragment>
                 <Router>
                     <Switch>
                         <Route path='/' exact render={() => <Page><Home /></Page>} />
                         <Route path='/login' render={() => <Page><Login /></Page>} />
-                        <Route path='/search/location' render={() => <Page><LocationReportSearch /></Page>} />
-                        <Route path='/search/region' render={() => <Page><RegionSummarySearch /></Page>} />
-                        <Route path='/submit/location' render={() => <Page><LocationReportSubmit /></Page>} />
-                        <Route path='/submit/region' render={() => <Page><RegionSummarySubmit /></Page>} />
-
-                        <Route path='/admin/locations' exact render={() => <Page><Entities api='locations' key='locations'/></Page>} />
-                        <Route path='/admin/regions' exact render={() => <Page><Entities api='regions' key='regions'/></Page>} />
-                        <Route path='/admin/users' exact render={() => <Page><Entities api='users' key='users'/></Page>} />
-
-                        <Route path='/admin/locations/add' render={() => <Page><LocationForm/></Page>} />
-                        <Route path='/admin/regions/add' render={() => <Page><RegionForm/></Page>} />
-                        <Route path='/admin/users/add' render={() => <Page><UserForm/></Page>} />
-
-                        <Route path='/admin/locations/edit/:id' render={props => <Page><LocationForm id={props.match.params.id} /></Page>} />
-                        <Route path='/admin/regions/edit/:id' render={props => <Page><RegionForm id={props.match.params.id} /></Page>} />
-                        <Route path='/admin/users/edit/:id' render={props => <Page><UserForm id={props.match.params.id} /></Page>} />
-
-                        <Route path='/message' render={() => <Page><TodaysCodePage /></Page>} />
-                        <Route path='/profile' exact render={() => <Page><Profile /></Page>} />
-                        <Route path='/profile/edit/callsign' render={() => <Page><EditProfileCallSign /></Page>} />
-                        <Route path='/profile/edit/picture' render={() => <Page><EditProfilePicture /></Page>} />
+                        {this.state.isLoggedIn ? this.renderLoggedIn(): null}
```

Comment:
This can be simplified to `{this.state.isLoggedIn && this.renderLoggedIn()}`.

Not super important but people tend to use this instead of having to handle the return null using `... : null`